File: Methodo/C/cle.c
#include <stdio.h>
#include <string.h>

int main(int argc, char const *argv[])
{
	char nb, nb2, nb3, nb4;
	//int tab[4] = {10, 20, 30, 40};

	printf("Entrez une valeur :"); scanf("%c%c%c%c", &nb, &nb2, &nb3, &nb4); //scanf("%c", &ch2); scanf("%c", &nb3); 


	switch(nb){
		case '1':
			switch(nb2){
				case '0':
					printf("nombre 100\n");
					break;

				default:
					printf("les cas précédents sont faux\n");
					break;
			}
			printf("nombre dans la liste\n");
			break;

		case 'a':
			printf("chaine a\n");
			break;

		case '4':
			switch(nb2){
				case '2':
					printf("nombre dans la liste\n");
					break;
			}
			break;

		default:
			printf("les cas précédents sont faux\n");
			break;

	}

	return 0;
}

File: Methodo/Perl/cle.pl
#!/usr/local/bin/perl
use Switch;
print "entrez une valeur : \n";
$nb = <STDIN>; chomp $nb;
@tab = (10, 20, 30, 40);
%cles = ('key1' => 10, 'key2' => 20);
switch($nb){
	case 10	{ print "nombre 100\n" }
	case "a"			{ print "chaine a\n" }
	case [1..10,42]		{ print "nombre dans la liste\n" }
	case (\@tab)		{ print "nombre dans la liste\n" }
	case (\%cles)		{ print "clef dans la table de hachage\n" } 
	else 				{ print "les cas précédents sont faux\n" }
}

